### PR TITLE
feat: directives and pipes

### PR DESCRIPTION
**Pull request by 6ypmac**
- custom status directive for courses is created
- ngIf added for no data message
- functionality for topRate is added (ngClass)
- uppercase built-in pipe is added for course name
- custom duration pipe is created
- date built-in pipe is added for course date
- filter for courses is added for search field

### CODE REVIEW

File: angular-video-course/src/app/app.module.ts [hunk 1]
```diff
@@ -5,18 +5,32 @@ import { FormsModule } from '@angular/forms';
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 import { CoreModule } from './core/core.module';
+import { CoursesPageComponent } from './courses-page/courses-page.component';
+import { CourseComponent } from './courses-page/course/course.component';
+import { SearchComponent } from './courses-page/search/search.component';
```
Comment (ezhylin) [inline angular-video-course/src/app/app.module.ts]: add index.ts to courses-page directory to export all component in the only import
For example https://medium.com/@balramchavan/smarter-way-to-organize-import-statements-using-index-ts-file-s-in-angular-c685e9d645b7

File: angular-video-course/src/app/app.module.ts [hunk 2]
```diff
@@ -5,18 +5,32 @@ import { FormsModule } from '@angular/forms';
...
+import { BorderDecoratorDirective } from './directives/border-decorator.directive';
+import { TransformMinutesPipe } from './pipes/transform-minutes.pipe';
```
Comment (ezhylin) [inline angular-video-course/src/app/app.module.ts]: Also, you can export all the pipes, in the same way as I mentioned above - using index.ts

Comment (6ypmac) [reply angular-video-course/src/app/app.module.ts]: done

File: angular-video-course/src/app/app.module.ts (hunk 1 above)
Comment (6ypmac) [reply angular-video-course/src/app/app.module.ts]: done

**Pull request merged**